plugins/postcss-todo-or-die/src/match/issue-open.ts

```diff
@@ -0,0 +1,104 @@
+import https from 'https';
+import path from 'path';
+
+export async function matchIssueOpenCondition(repository: string, issue: number, cache: Map<string, string>): Promise<string | true | undefined> {
+	const cacheKey = `github-issue-open:${repository}:${issue}`;
+	let state;
+
+	if (!cache.has(cacheKey)) {
+		try {
+			state = await fetch(repository, issue);
+		} catch (_) {
+
+			// Any network errors are ignored.
+			// We do not want builds to fail when GitHub is offline.
+			return;
+		}
+	}
+
+	if (state === 'closed') {
+		cache.set(cacheKey, state);
+		return `Died because issue ${issue} in ${repository} was closed`;
+	}
+
+	if (state === 'open') {
+		cache.set(cacheKey, state);
+		return true;
+	}
+
+	return;
+}
+
+function fetch(repository: string, issue: number) {
+	return new Promise<string>((resolve, reject) => {
+		let rejected = false;
+
+		const url = new URL(
+			path.join('repos', repository, 'issues', issue.toString()),
+			'https://api.github.com/',
+		);
+
+		const options = {
+			headers: {
+				'User-Agent': 'PostCSS TODO or Die',
+			},
+			timeout: 15000,
+		};
+
+		if (process.env.GITHUB_TOKEN) {
+			options.headers['Authorization'] = `Bearer ${process.env.GITHUB_TOKEN}`;
+		}
+
+		const request = https.request(url, options, (response) => {
+			if (rejected) {
+				return;
+			}
+
+			if (response.statusCode !== 200) {
+				rejected = true;
+				reject(new Error('Failed to fetch issue status'));
+				return;
+			}
```

Comment:
I was wondering what happens if you don't have visibility to an issue because you're lacking the token or whatever I feel this should at least be surfaced somehow since you might think it's working but perhaps it isn't

Reply:
I am going to remove the "issue open" feature entirely :)

`todo-or-die` for Rust has this feature, but you compile Rust much less frequently than you build CSS during development. When working in Rust or Go I don't use a file watcher to re-compile.

Doing so many builds so frequently is bound to cause false errors because it depends on the network. You also should be able to work offline without seeing errors.

I think it makes more sense to remove this, and maybe later add something that allows people to easily add custom checks.